### PR TITLE
Add null support to bitmap index.

### PR DESCRIPTION
**Pull request by yueawang**
## What changes were proposed in this pull request?

Add null value support to bitmap index.
1. Add null value index read & write.
2. Add test cases for normal and corner cases.

## How was this patch tested?
mvn test passes.

### CODE REVIEW

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BitMapScanner.scala [hunk 1]
```diff
@@ -224,17 +242,22 @@ private[oap] case class BitMapScanner(idxMeta: IndexMeta) extends IndexScanner(i
...
+          if (startIdx == -1 || endIdx == -1) {
+            // range not fond in cur bitmap, return empty for performance consideration
+            Seq.empty[ImmutableRoaringBitmap]
+          } else {
+            val startIdxOffset = getStartIdxOffset(bmOffsetListCache, 0L, startIdx)
+            val curPosition = startIdxOffset - bmEntryListOffset
+            getDesiredBitmaps(bmEntryListCache.toArray, curPosition, startIdx, endIdx)
+          }
+        case range if range.isNullPredicate =>
+          getDesiredBitmaps(bmNullListCache.toArray, 0, 0, 0)
```
Comment (lee-lei) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/index/BitMapScanner.scala]: Can we avoid to use toArray in line 258? The latest master doesn't use toArrary in line 255. Thanks.

**Comment by lee-lei** [comment]
Please help rebase on the latest master. Thanks.

**Comment by adrian-wang** [comment]
LGTM, thanks!
